src/cls/rgw/cls_rgw.cc

```diff
@@ -446,7 +446,7 @@ int rgw_bucket_list(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
         continue;
       }
 
-      if (!op.list_versions && !entry.is_visible()) {
+      if (!op.list_versions && !entry.is_visible() || (!op.list_versions && op.start_obj.name == key.name)) {
```

Comment:
can simplify this a bit, and a comment would help:
```c++
      // filter out noncurrent versions, delete markers, and initial marker
      if (!op.list_versions && (!entry.is_visible() || op.start_obj.name == key.name)) {
```

Reply:
hi , @cbodley ,i updated the code :）